lms/djangoapps/course_api/blocks/api.py

```diff
@@ -17,7 +17,9 @@ def get_blocks(
         requested_fields=None,
         block_counts=None,
         student_view_data=None,
-        return_type='dict'
+        return_type='dict',
+        block_types=None,
```

Comment:
The `block_types` parameter is a filter.  To be consistent with other filters implemented in this API, we should introduce a new `BlockTypesFilterTransformer`.  The new transformer would then be processed along with other Blocks API transformers, as done [here](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/course_api/blocks/transformers/blocks_api.py#L60).

Regarding conflicting with `block_counts`, I don't see it that way.  `block_counts` returns an aggregate count of the number of blocks within each subtree.  Whereas `block_types` is a filter, specifying exactly which types of blocks to return.  It's possible that `block_counts` may contain additional types that are not included in `block_types`.  In that case, the counts for those should still be returned, but the blocks themselves wouldn't.

In order to implement it as such, we would order the `block_counts` Transformer *before* the `block_types` Transformer.  Since the latter would filter out blocks before they are counted.